### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by schlosna**
## General
**Before this PR**:

Dependency upgrade excavator https://github.com/palantir/atlasdb/pull/6543 is blocked by several dependencies. 

**After this PR**:

==COMMIT_MSG==
Upgrade dependencies
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

**Review by carterkozak — APPROVED** [review]

**Comment by carterkozak** [comment]
huh, this is failing with `No tests found for given includes: [com.palantir.atlasdb.timelock.stress.*](filter.includeTestsMatching)`

File: versions.lock [hunk 1]
```diff
@@ -358,21 +359,23 @@ org.glassfish.grizzly:grizzly-framework:2.4.4 (3 constraints: bc31e6d7)
...
-org.junit.jupiter:junit-jupiter-engine:5.4.2 (1 constraints: 92108bb2)
-org.junit.platform:junit-platform-commons:1.4.2 (2 constraints: d5202f4a)
-org.junit.platform:junit-platform-engine:1.4.2 (2 constraints: ed22c407)
+org.junit.jupiter:junit-jupiter:5.9.1 (2 constraints: 923d8556)
+org.junit.jupiter:junit-jupiter-api:5.9.1 (4 constraints: 0740843c)
+org.junit.jupiter:junit-jupiter-engine:5.9.1 (2 constraints: 9d1e4185)
+org.junit.jupiter:junit-jupiter-params:5.9.1 (1 constraints: 0c0ee13b)
+org.junit.platform:junit-platform-commons:1.9.1 (2 constraints: dd200f4b)
+org.junit.platform:junit-platform-engine:1.9.1 (2 constraints: f1229e08)
 org.junit.platform:junit-platform-launcher:1.4.2 (1 constraints: 8e1077b2)
```
Comment (tpetracca) [inline versions.lock]: feels weird that this dep isn't getting bumped with the rest of the `org.junit.platform:junit-platform-*` deps

Comment (tpetracca) [reply versions.lock]: probably warrants a glob in versions.props?

**Comment by tpetracca** [comment]
>huh, this is failing with No tests found for given includes: [com.palantir.atlasdb.timelock.stress.*](filter.includeTestsMatching)

I'm guessing one of the junit version bumps is doing this and failing situations where previously running no tests was okay. If you look on develop these have zero un-ignored tests as well...
- test-suite-6 failure: https://output.circle-artifacts.com/output/job/da92e75b-9726-4446-b980-9a3e27b71cdc/artifacts/0/home/circleci/artifacts/junit/timelock-server/suiteTest/packages/com.palantir.atlasdb.timelock.suite.html
- test-suite-7 failure: https://output.circle-artifacts.com/output/job/fe3fe397-ec03-490a-b789-2d71fccc3815/artifacts/0/home/circleci/artifacts/junit/timelock-server/stressTest/index.html

**Comment by tpetracca** [comment]
We should confirm this is the expected state for both of these with @LucasIME or @jeremyk-91, and if so we'll want to add `setFailOnNoMatchingTests(false)` where appropriate, like we have [here](https://github.com/palantir/atlasdb/blob/develop/atlasdb-ete-tests/build.gradle#L186-L188).

**Comment by schlosna** [comment]
> We should confirm this is the expected state for both of these with @LucasIME or @jeremyk-91, and if so we'll want to add `setFailOnNoMatchingTests(false)` where appropriate, like we have [here](https://github.com/palantir/atlasdb/blob/develop/atlasdb-ete-tests/build.gradle#L186-L188).

Looking at the timelock-server `stressTest` tests, they all use JUnit 4 test suites and class rules, so we need to use the JUnit vintage runtime for them to actually run in JUnit 5 jupiter.  

https://github.com/palantir/atlasdb/blob/fc2310678a8f6f3fe225c934e26c908d84c9a86b/timelock-server/src/stressTest/java/com/palantir/atlasdb/timelock/stress/DbTimeLockSingleLeaderPaxosStressTests.java#L21-L40

https://github.com/palantir/atlasdb/blob/fc2310678a8f6f3fe225c934e26c908d84c9a86b/timelock-server/src/stressTest/java/com/palantir/atlasdb/timelock/stress/MultiLeaderPaxosStressTests.java#L21-L35

https://github.com/palantir/atlasdb/blob/fc2310678a8f6f3fe225c934e26c908d84c9a86b/timelock-server/src/stressTest/java/com/palantir/atlasdb/timelock/stress/SingleLeaderPaxosStressTests.java#L21-L35

**Pull request merged**